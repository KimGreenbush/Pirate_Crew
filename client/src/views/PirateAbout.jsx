import React, {useState, useEffect} from "react"
import { Link } from "@reach/router"
import axios from "axios"

const Show = ({id}) => {
    const [pirate, setPirate] = useState({})
    const [peg, setPeg] = useState()
    const [patch, setPatch] = useState()
    const [hook, setHook] = useState()

    useEffect(() => {
        axios.get(`http://localhost:9001/api/pirate/${id}`)
            .then(res => {
                setPirate(res.data)
                setPeg(res.data.peg)
                setPatch(res.data.patch)
                setHook(res.data.hook)
            })
            .catch(err => console.log("Error: ", err))
    }, [])

    const changePeg = () => {
        setPeg(!peg)
        axios.put(`http://localhost:9001/api/pirate/${id}`, {peg: !peg })
            .then((res) => {
                console.log(res)
            })
            .catch((err) => {
				const errorRes = err.response.data
				console.log(errorRes)
			})
    }

    const changePatch = () => {
        setPatch(!patch)
        axios.put(`http://localhost:9001/api/pirate/${id}`, {patch: !patch })
            .then((res) => {
                console.log(res)
            })
            .catch((err) => {
				const errorRes = err.response.data
				console.log(errorRes)
			})
    }

    const changeHook = () => {
        setHook(!hook)
        axios.put(`http://localhost:9001/api/pirate/${id}`, {hook: !hook })
            .then((res) => {
                console.log(res)
            })
            .catch((err) => {
				const errorRes = err.response.data
				console.log(errorRes)
			})
    }

    return (
        <div className="pirate-about">
            <Link to="/pirate"><button>Back to All Pirates</button></Link>
            <h1>{pirate.name}</h1>
            <div className="column">
                <figure>
                    <img src={pirate.image} alt="pirate" />
                    <figcaption>"{pirate.phrase}" - {pirate.name}</figcaption>
                </figure>
            </div>
            <div className="column">
                <h2>About: </h2>
                <p>Position: {pirate.position}</p>
                <p>Treasures: {pirate.chests}</p>
                <p>
                    Peg Leg: {peg ? "Yes" : "No"}
                    <button onClick={changePeg}>{peg ? "No" : "Yes"}</button>
                </p>
                <p>
                    Eye Patch: {patch ? "Yes" : "No"}
                    <button onClick={changePatch} >{patch ? "No" : "Yes"}</button>
                </p>
                <p>
                    Hook Hand: {hook ? "Yes" : "No"}
                    <button onClick={changeHook} >{hook ? "No" : "Yes"}</button>
                </p>
            </div>
        </div >
    )
}

export default Show