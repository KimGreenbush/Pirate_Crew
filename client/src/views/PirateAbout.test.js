/**
 * @jest-environment jsdom
 */

import React from "react";
import PirateAbout from "./PirateAbout";
import renderer from "react-test-renderer";

test("renders PirateAbout without crashing", () => {
	const component = renderer.create(<PirateAbout />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
