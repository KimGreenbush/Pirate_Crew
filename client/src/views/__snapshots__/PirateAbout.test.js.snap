// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PirateAbout without crashing 1`] = `
<div
  className="pirate-about"
>
  <a
    href="/pirate"
    onClick={[Function]}
  >
    <button>
      Back to All Pirates
    </button>
  </a>
  <h1 />
  <div
    className="column"
  >
    <figure>
      <img
        alt="pirate"
      />
      <figcaption>
        "
        " - 
      </figcaption>
    </figure>
  </div>
  <div
    className="column"
  >
    <h2>
      About: 
    </h2>
    <p>
      Position: 
    </p>
    <p>
      Treasures: 
    </p>
    <p>
      Peg Leg: 
      No
      <button
        onClick={[Function]}
      >
        Yes
      </button>
    </p>
    <p>
      Eye Patch: 
      No
      <button
        onClick={[Function]}
      >
        Yes
      </button>
    </p>
    <p>
      Hook Hand: 
      No
      <button
        onClick={[Function]}
      >
        Yes
      </button>
    </p>
  </div>
</div>
`;
